src/Simulation/Simulators/SparseSimulator/SparseSimulator.cs

```diff
@@ -34,16 +38,51 @@ public partial class SparseSimulator : CommonNativeSimulator
                randomNumberGeneratorSeed,
                disableBorrowing)
         {
-            Id = init_cpp((QubitIdType)numQubits);
+            // Init the telemetry first, to catch/log the construction issues:
+            TelemetryConfig = new TelemetryManagerConfig()
+            {
+                AppId = this.Name,
+                SendTelemetryInitializedEvent = true,
+                SendTelemetryTearDownEvent = true,
+            };
+            TelemetryManagerDisposer = TelemetryManager.Initialize(TelemetryConfig);
+
+            Microsoft.Applications.Events.EventProperties eventProperties = new Applications.Events.EventProperties()
+                {
+                    Name = "Quantum.SparseSim.Constructed",
+                };
+            eventProperties.SetProperty("throwOnReleasingQubitsNotInZeroState", throwOnReleasingQubitsNotInZeroState);
+            // (`SimulatorBase.Seed` is assigned to `randomNumberGeneratorSeed ?? Guid.NewGuid().GetHashCode()`):
+            eventProperties.SetProperty("randomNumberGeneratorSeed", Seed); 
+            eventProperties.SetProperty("disableBorrowing", disableBorrowing);
+            eventProperties.SetProperty("numQubits", numQubits);
```

Comment:
I believe the qubit manager by default allows to allocate more than numQubits, though it can be configured to fail if more qubits than the initially specified number is ever allocated. If we want to gain an understanding of the maximum number of qubits in use, we would need to track this number in the qubit manager, and send it only after the execution terminates.